java/common/exception/ErrorMessage.java

```diff
@@ -67,14 +67,64 @@
             new ErrorMessage(23, "Rule '%s' 'then' '%s': is trying to assign both an attribute type and a variable attribute value.");
     public static final ErrorMessage INVALID_RULE_THEN_VARIABLES =
             new ErrorMessage(24, "Rule '%s' 'then' variables must be present in rule 'when'.");
-    public static final ErrorMessage REDUNDANT_NESTED_NEGATION =
+    public static final ErrorMessage INVALID_REDUNDANT_NESTED_NEGATION =
             new ErrorMessage(25, "Invalid query containing redundant nested negations.");
     public static final ErrorMessage MISSING_COMPUTE_CONDITION =
             new ErrorMessage(26, "Missing condition(s) for 'compute '%s''. The required condition(s) are: '%s'.");
     public static final ErrorMessage INVALID_COMPUTE_METHOD_ALGORITHM =
             new ErrorMessage(27, "Invalid algorithm for 'compute '%s''. The accepted algorithm(s) are: '%s'.");
     public static final ErrorMessage INVALID_COMPUTE_ARGUMENT =
             new ErrorMessage(28, "Invalid argument(s) 'compute %s using %s'. The accepted argument(s) are: '%s'.");
+    public static final ErrorMessage INVALID_SORTING_ORDER =
```

Comment:
Do we want to add this many error messages @haikalpribadi ?

Reply:
@lolski changed most of them to ILLEGAL_GRAMMAR per Haikal's request